eks-worker.tf

```diff
@@ -44,7 +44,7 @@ module "aws_eks_managed_node_groups" {
   cluster_primary_security_group_id    = module.aws_eks.cluster_primary_security_group_id
   tags                                 = module.eks_tags.tags
 
-  depends_on = [module.aws_eks, kubernetes_config_map.aws_auth]
+  depends_on = [kubernetes_config_map.aws_auth]
```

Comment:
Do we need this?  Is it giving the same response time as implicit dependency with merge tags?

Reply:
pretty much same results either way:
- With explicit dependency (depends_on): 1min 18seconds https://github.com/aws-samples/aws-eks-accelerator-for-terraform/runs/5190500712?check_suite_focus=true
- With implicit dependency (via tags): 1min 5seconds  https://github.com/aws-samples/aws-eks-accelerator-for-terraform/runs/5190867428?check_suite_focus=true

The implicit dependency is on different branch (`fix/depends-on-implicit-test` , was checked out from this PR branch) for comparing and testing purposes.